caluma/caluma_form/domain_logic.py

```diff
@@ -151,6 +152,26 @@ def post_save(answer: models.Answer) -> models.Answer:
         # TODO emit events
         return answer
 
+    @staticmethod
+    def update_calc_dependents(answer):
+        if not answer.question.calc_dependents:
+            return
+
+        root_doc = answer.document.family
+        root_doc = (
+            models.Document.objects.filter(pk=answer.document.family_id)
+            .prefetch_related(
+                *utils.build_document_prefetch_statements(prefetch_options=True)
+            )
+            .first()
+        )
+        struc = structure.FieldSet(root_doc, root_doc.form)
```

Comment:
How about we refactor this dance into the fieldset itself? So this becomes something along the lines of

```py
struc = structure.FieldSet.with_prefetch(root_doc, root_doc.form)
```

The implementation could then be something like

```py
class FieldSet(...):
    ...
    @classmethod
    def with_prefetch(cls, root_doc, form=None):
        root_doc = (
            models.Document.objects.filter(pk=answer.document.family_id)
            .prefetch_related(
                *utils.build_document_prefetch_statements(prefetch_options=True)
            )
            .first()
        )
        return cls(root_doc,form)
```

Reply:
Changed the callsite to `prefetch_document` instead, as discussed.